core/src/main/java/org/testcontainers/containers/ContainerLifecycleHooks.java

```diff
@@ -0,0 +1,11 @@
+package org.testcontainers.containers;
+
+interface ContainerLifecycleHooks {
+    ContainerLifecycleHooks EMPTY = new ContainerLifecycleHooks() {};
+
+    default void configure() {}
+
+    default void containerIsStarting(boolean reused) {}
+
+    default void containerIsStarted(boolean reused) {}
```

Comment:
Those were initially added to be used in Kafka and MongoDB implementations. `configure` is no longer used to avoid breaking changes in Kafka.

There is a proposal about the lifecycle hooks:

* Before the container is created
* After the container is created
* Before the container is started
* After the container is started
* Before the container is stopped
* After the container is stopped
* Before the container is terminated
* After the container is terminated

but `After the container is created` and `Before the container is started` would be the same. Although, I'm in favor of consistency, this could be confusing IMO. Also, I think we will remain with termination actions instead of stop.

Better to discuss it later, wdyt?

Reply:
I would prefer to remove the lifecycle topics from the scope of this PR if possible. From a recent discussion in tc-go we also learned, there seems to be a use case for hooks around "container ready" (i.e. wait strategy finished).